File: JJB/Programmers/231203/소인수분해.cpp
#include <string>
#include <vector>
#include <cmath>

using namespace std;

vector<int> solution(int n) 
{
    vector<int> answer;
    
    for(int i = 2; i <= n; i++)
    {
        if(n==1)
            break;
        
        else if(n%i != 0)
            continue;
        
        while(n%i == 0)
            n /= i;
        answer.push_back(i);
    }
    
    return answer;
}

File: JJB/Programmers/231203/소인수분해.py
import math

def solution(n):
    answer = []
    
    for divider in range(2, n+1):
        if n==1:
            break
            
        if n%divider != 0:
            continue
        
        while n%divider == 0:
            n //= divider
        answer.append(divider)
    
    return answer